inst/include/dust/dust.hpp

```diff
@@ -931,15 +1009,18 @@ void run_particles(size_t step_start, size_t step_end, size_t n_particles,
     // TODO: this needs work before moving to the device, but it might
     // not be that bad in practice. We'll need some extra code to deal
     // with the blocks (before the loop) too.
-    dust::shared_ptr<T> p_shared = shared[i / n_particles_each];
+    const int j = i / n_particles_each;
+    const int * p_shared_int = shared_int + j * n_shared_int;
+    const real_t * p_shared_real = shared_real + j * n_shared_real;
```

Comment:
This is nice and should easily be changed to the CUDA pattern I was thinking of

Reply:
I think it will work as-is in fact, but we can speed it up by loading into L1 shared if it will fit

Reply:
And makes it easy to make conditional on n_particles_each too